src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/LegacyCSProjPackageReferenceProject.cs

```diff
@@ -373,7 +373,7 @@ private static LibraryDependency ToPackageLibraryDependency(LegacyCSProjPackageR
             {
                 LibraryRange = new LibraryRange(
                     name: item.Name,
-                    versionRange: new VersionRange(new NuGetVersion(item.Version), originalString: item.Version),
+                    versionRange: VersionRange.Parse(item.Version),
```

Comment:
It will throw the same error stating not a valid version string... And I think that will be the right behavior since user actually went into csproj and manually udpated version string so throwing that error makes sense so that user can put the right version string there.